src/librustc_middle/mir/interpret/mod.rs

```diff
@@ -549,18 +549,20 @@ pub fn write_target_uint(
     mut target: &mut [u8],
     data: u128,
 ) -> Result<(), io::Error> {
-    let len = target.len();
     match endianness {
-        Endian::Little => target.write_uint128::<LittleEndian>(data, len),
-        Endian::Big => target.write_uint128::<BigEndian>(data, len),
-    }
+        Endian::Little => target.write(&data.to_le_bytes())?,
```

Comment:
I believe it's not promised that all bytes are written if `write_all` is not used.

Reply:
I believe the data is encoded as u128 but actually will experience a lossy conversion if the desired uint at the end point is only e.g. u64-width (such as with a usize on x86_64), and thus write_all is actually not the desired behavior.

Reply:
```rust
    #[inline]
    fn write(&mut self, data: &[u8]) -> io::Result<usize> {
        let amt = cmp::min(data.len(), self.len());
        let (a, b) = mem::replace(self, &mut []).split_at_mut(amt);
        a.copy_from_slice(&data[..amt]);
        *self = b;
        Ok(amt)
    }
```

This is desired behaviour. I can include an assert_eq! on write's return and the target's length, but doing so is just double-checking that `impl Write for &mut [u8]` is correct.

Reply:
Verified that write_all hits WriteZero errs, same story with read_exact, I can do what byteorder did and `write_all[0..len]` but it still seems spiritually incorrect to call a looping method when I know it will only write once ever, and

```rust
    #[inline]
    fn write_all(&mut self, data: &[u8]) -> io::Result<()> {
        if self.write(data)? == data.len() {
            Ok(())
        } else {
            Err(Error::new(ErrorKind::WriteZero, "failed to write whole buffer"))
        }
    }
```
introduce an Err only to snuff it.

I think I can trust Rust to write 0~16 bytes consistently in a single invocation. :^)